netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -201,25 +200,42 @@ public void onStreamRemoved(Http2Stream stream) {
   @Override
   public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
           throws Exception {
-    if (msg instanceof CreateStreamCommand) {
-      createStream((CreateStreamCommand) msg, promise);
-    } else if (msg instanceof SendGrpcFrameCommand) {
-      sendGrpcFrame(ctx, (SendGrpcFrameCommand) msg, promise);
-    } else if (msg instanceof CancelClientStreamCommand) {
-      cancelStream(ctx, (CancelClientStreamCommand) msg, promise);
-    } else if (msg instanceof RequestMessagesCommand) {
-      ((RequestMessagesCommand) msg).requestMessages();
-    } else if (msg instanceof SendPingCommand) {
-      sendPingFrame(ctx, (SendPingCommand) msg, promise);
-    } else if (msg instanceof GracefulCloseCommand) {
-      gracefulClose(ctx, (GracefulCloseCommand) msg, promise);
-    } else if (msg instanceof ForcefulCloseCommand) {
-      forcefulClose(ctx, (ForcefulCloseCommand) msg, promise);
-    } else if (msg == NOOP_MESSAGE) {
-      ctx.write(Unpooled.EMPTY_BUFFER, promise);
-    } else {
-      throw new AssertionError("Write called for unexpected type: " + msg.getClass().getName());
+    UnionCommand cmd = (UnionCommand) msg;
+    switch (cmd.type()) {
```

Comment:
Tricky, what happens if there is an exception that is thrown after the object has been recycled in the sub method?  That would result in a double free.  

It is much safer to just let the jvm pick it up, assuming the Handle is okay to GC.